sound/soc/codecs/rt715-sdca.c

```diff
@@ -286,7 +288,22 @@ static int rt715_sdca_put_volsw(struct snd_kcontrol *kcontrol,
 		if (err < 0)
 			return err;
 	}
-
+#if IS_ENABLED(CONFIG_DELL_PRIVACY)
+	/* Privacy LED Trigger State Changed by muted/unmute switch */
+	if (mc->invert) {
+		val0 = ucontrol->value.integer.value[0];
+		val1 = ucontrol->value.integer.value[1];
+		if (val0 == 1 && val1 == 1) {
+			rt715->micmute_led = LED_OFF;
+			ledtrig_audio_set(LED_AUDIO_MICMUTE,
+					rt715->micmute_led ? LED_ON : LED_OFF);
+		} else if (val0 == 0 && val1 == 0) {
```

Comment:
You handle `val0 == 1 && val1 == 1` and `val0 == 0 && val1 == 0` only. What about other cases? IMHO, the mute led should be off if any channel of mic is on. My suggestion will be
```patch
+               if (ucontrol->value.integer.value[0] || ucontrol->value.integer.value[1])
+                       rt715->micmute_led = LED_OFF;
+               else
+                       rt715->micmute_led = LED_ON;
```

Reply:
> You handle `val0 == 1 && val1 == 1` and `val0 == 0 && val1 == 0` only. What about other cases? IMHO, the mute led should be off if any channel of mic is on. My suggestion will be
> 
> ```diff
> +               if (ucontrol->value.integer.value[0] || ucontrol->value.integer.value[1])
> +                       rt715->micmute_led = LED_OFF;
> +               else
> +                       rt715->micmute_led = LED_ON;
> ```

Hi @bardliao 
I have thought this before, just a fews things to confirm.
 * the mic mute  will mute / unmute all the channel or not ?
 * I add some debug log, the value[0] and value[1], will set many times when doing the mute process.
    in most common case ,the user don`t use single channel, the last setting call to codec driver will set value[0] and value[1] with 0 or 1 at the same time.

* if i use  "value[0]  | value[1] ", the ledtrig_audio_set will be called  3 or 4 times for one time mute ops.

Reply:
> Hi @bardliao
> I have thought this before, just a fews things to confirm.
> 
> * the mic mute  will mute / unmute all the channel or not ?

The mic mute key should mute/unmute all the channels.

> * I add some debug log, the value[0] and value[1], will set many times when doing the mute process.

IMHO, it depends on how user space handle the mute key event. We should be able to set the value[0] and [1] only once. 
BTW, you can do nothing in ledtrig_audio_set() if the LED status is not changed.

>   in most common case ,the user don`t use single channel, the last setting call to codec driver will set value[0] and value[1] with 0 or 1 at the same time.
> * if i use  "value[0]  | value[1] ", the ledtrig_audio_set will be called  3 or 4 times for one time mute ops.

The point is that if one of value[0] or value[1] is 1, it means mic is NOT muted.